### ISSUE TITLE
Jobs shortcode needs an option to filter by user.

### ISSUE THREAD

**Issue by greggh** [Automattic/WP-Job-Manager#1892]
The jobs shortcode has quite a lot of functionality, but it is literally missing the only functionality I actually need.

I need the ability to place jobs on a users profile. For my purposes, it is in BuddyPress. Creating a tab/profile page for them is easily done. But putting the jobs shortcode in, there is no way to get only jobs created by this specific user.

Ideally, I should be able to add a user=2, or whatever ID I am currently viewing, into the shortcode to only show that users jobs. A secondary enhancement that would be quite amazing would be an empty_message="No results found." added to the shortcode so I could set the text shown in that instance.

Many themes properly style the HTML output of the jobs shortcode, so it is the best possible place to put this filter.

**Comment by greggh** [comment]
I saw in the last update that support for Author was added to the rest api. Does that get us closer to adding this into the shortcode?

**Comment by datdudejibril** [comment]
I needed this functionality also. There is a plugin referred by support that is supposed to work but it doesn't. So I'm custom coding a shortcode to display jobs based on the user profile currently being viewed.

**Comment by Dazman123** [comment]
I also need a solution to this. Why is there no way to display a specific users jobs??

**Comment by Dazman123** [comment]
> I needed this functionality also. There is a plugin referred by support that is supposed to work but it doesn't. So I'm custom coding a shortcode to display jobs based on the user profile currently being viewed.

Have you had any luck with custom coding this? And if so do you mind sharing your solution?

**Comment by BadOmn83** [comment]
Please implement this soon.

**Comment by nicolascuny** [comment]
As recommended on the plugin support forum, I am adding my name here. I need this filter too.

**Comment by pierrealexislebrun** [comment]
I agree, that would be a great feature to add to the plugin in the future

**Comment by NicoHood** [comment]
Coming here 5 Years later this is still an issue, would love to see this implemented.

**Comment by masteradhoc** [comment]
Hey everyone

i prepared a PullRequest in https://github.com/Automattic/WP-Job-Manager/pull/2935 to bring this feature into WPJM. Please give it a try and let me know if it works for you as well.
